defopt.py

```diff
@@ -118,8 +118,23 @@ def _create_parser(funcs, *args, **kwargs):
     return parser
 
 
-class _Formatter(RawTextHelpFormatter, ArgumentDefaultsHelpFormatter):
-    pass
+class _Formatter(RawTextHelpFormatter):
+
+    # modified from ArgumentDefaultsHelpFormatter to add type information, and
+    # remove defaults for varargs (`ZERO_OR_MORE`).
+    def _get_help_string(self, action):
+        info = []
+        if action.type is not None and '%(type)' not in action.help:
+            info.append('%(type)s')
+        if (action.default != argparse.SUPPRESS
```

Comment:
The intent of this code is not to be a string comparison. `action.default` may not be a string, and `SUPPRESS` only happens to be a string in implementation. Unlike some of the other constants, the documentation for argparse does not describe what its value is. The documentation tells you to use `nargs='?'` - the value of `argparse.OPTIONAL` *must* be `'?'`. Using `default='==SUPPRESS=='` instead of `default=argparse.SUPPRESS` (if it actually works) is violating the API - it could equally be well be implemented as `SUPPRESS = object()`. In terms of edge cases where this matters, perhaps your default is some object that is deliberately supposed to compare equal to anything, in which case this must compare with `is not` to avoid the default being suppressed.